code/ship/ship.cpp

```diff
@@ -13552,6 +13559,24 @@ void ship_model_replicate_submodels(object *objp)
 			model_replicate_submodel_instance(pm, pmi, psub->turret_gun_sobj, pss->flags );
 		}
 	}
+
+	// Keep other movable submodels in sync
+	model_iterate_submodel_tree(pm, pm->detail[0], [&](int submodel, int level, bool isLeaf)
+		{
+			auto sm = &pm->submodel[submodel];
+
+			// skip submodels that belong to subsystems since we already updated them above
+			// (Turrets are tricky because they have two submodels per subsystem, but because the subsystem index is copied
+			// when the model is read, both submodels are correctly skipped here.)
+			if (sm->subsys_num >= 0)
+				return;
+
+			// the only non-subsystem submodels that need to be updated are the ones that can move
+			if (!sm->flags[Model::Submodel_flags::Can_move])
```

Comment:
This assumption is not true anymore.
The $triggered property is no longer required due to the way that animations step over time.
Hence, an animated submodel might not have this flag set.

Reply:
That flag needs to be set though.  Rotating subsystems set that flag, and so will translating subsystems once that's finished.  Plenty of places in the code use that flag to determine whether they should treat submodels as static or moveable: dockpoints, thrusters, etc.

In fact I just realized the collision code needs to be fixed to use it too; it currently uses an alternate method that won't capture all situations.

Reply:
Added the flag to the collision code.

Reply:
#3972 addresses the issue that not all animated subsystems can have Can_move